KlarnaOnDemandTests/KODOriginProofSpec.m

```diff
@@ -35,17 +28,29 @@ + (NSString *)timestamp;
     [[dataDic[@"amount"] should] equal:[NSNumber numberWithInt:3600]];
     [[dataDic[@"currency"] should] equal:@"SEK"];
     [[dataDic[@"user_token"] should] equal:@"my_token"];
-    [[dataDic[@"timestamp"] should] equal:[KODOriginProof timestamp]];
+    [[dataDic[@"id"] should] matchPattern:@"[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}"];
     [[originProofDic[@"signature"] should] equal:@"my_signature"];
   });
   
-});
-
-describe(@".timestamp", ^{
-  NSDate *now = [NSDate dateWithTimeIntervalSince1970:0];
-  [NSDate stub:@selector(date) andReturn:now];
-
-  [[[KODOriginProof timestamp] should] equal:@"1970-01-01T02:00:00+02:00"];
+  it(@"should generate different id for each order", ^{
+    NSString *originProofA = [KODOriginProof generateWithAmount:3600 currency:@"SEK" userToken:@"my_token"];
+    NSData *decodedriginProofA = [[NSData alloc] initWithBase64EncodedString:originProofA options:0];
+    NSDictionary *originProofDicA = [NSJSONSerialization JSONObjectWithData: decodedriginProofA
+                                                                   options: NSJSONReadingMutableContainers
+                                                                      error: nil];
+    NSData *dataA = [originProofDicA[@"data"] dataUsingEncoding:NSUTF8StringEncoding];
+    NSDictionary *dataDicA = [NSJSONSerialization JSONObjectWithData:dataA options:0 error:nil];
+    
+    NSString *originProofB = [KODOriginProof generateWithAmount:3600 currency:@"SEK" userToken:@"my_token"];
+    NSData *decodedriginProofB = [[NSData alloc] initWithBase64EncodedString:originProofB options:0];
+    NSDictionary *originProofDicB = [NSJSONSerialization JSONObjectWithData: decodedriginProofB
+                                                                    options: NSJSONReadingMutableContainers
+                                                                      error: nil];
+    NSData *dataB = [originProofDicB[@"data"] dataUsingEncoding:NSUTF8StringEncoding];
+    NSDictionary *dataDicB = [NSJSONSerialization JSONObjectWithData:dataB options:0 error:nil];
+    
+    [[dataDicA[@"id"] shouldNot] equal:dataDicB[@"id"]];
+  });
```

Comment:
You can create a helper method that gets the internal decoded data from the origin proof.